src/components/LoginForm.js

```diff
@@ -0,0 +1,137 @@
+import { useState, useEffect } from "react";
+import { useTransition, animated } from "react-spring";
+import { Link, useHistory } from "react-router-dom";
+import { login, validateEmail } from "../service.js";
+import Input from "./Input";
+
+const LoginForm = ({ login }) => {
+    const [user, setUser] = useState({
```

Comment:
Zamiast dwóch state `user` i `isValid` zrobiłbym 4 dla każdego pola. Będzie wtedy można pozbyć się obsługiwania `id` w inpucie, co uprości kod. 

Podobnie można uprościć obsługę errorów i zamiast używać booleanu valid, to dać array z errorami i w zależności czy jest pusty czy nie dodawać odpowiednie klasy i message.